File: Regular Expressions/Code/chapter1/1-4/recipe1-4.pl
#!/usr/bin/perl -w
use strict;

my $mystr = $ARGV[0] || die "You must supply a parameter!\n";
if ( $mystr =~ /Joh?n(athan)? Doe/) 
{
	print "Hello!  I've been looking for you!\n" 
}

File: Regular Expressions/Code/chapter1/1-4/recipe1-4.py
#!/usr/bin/python
import re
import sys

nargs = len(sys.argv)

if nargs > 1:
	mystr = sys.argv[1]

	r = re.compile(r'Joh?n(athan)? Doe', re.M)
	if r.search(mystr):
		print 'Here\'s Johnny!',
	else: 
		print 'Who?',

else:
	print 'I came here for an argument!',
